resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicExecutorFeatureStep.scala

```diff
@@ -54,7 +54,8 @@ private[spark] class BasicExecutorFeatureStep(
 
   private val memoryOverheadMiB = kubernetesConf
     .get(EXECUTOR_MEMORY_OVERHEAD)
-    .getOrElse(math.max((MEMORY_OVERHEAD_FACTOR * executorMemoryMiB).toInt,
+    .getOrElse(math.max(
+      (kubernetesConf.get(MEMORY_OVERHEAD_FACTOR).getOrElse(0.1) * executorMemoryMiB).toInt,
```

Comment:
This logic happens twice, and I'd rather see it in a utils or config class where if we update it in one place it will take effect everywhere.

Reply:
Logic is different since it takes in Driver vs. Executor configs to determine memory size. only similar logic is in `kubernetesConf.get(MEMORY_OVERHEAD_FACTOR).getOrElse(0.1)` which seems unnecessary to put into a Utils.

Reply:
Yeah so multiple yes, but since have two instances of this magic 0.1 constant I'd rather have it shared somewhere incase we go to update this in the future and don't get everywhere. Could also be a shared constant if we want instead rather than a getter for memory overhead factor, either way keeps us honest.